### ISSUE TITLE
fmt::format_to_n does not work with FMT_COMPILE-d format string

### ISSUE THREAD

**Issue by 0xb8** [fmtlib/fmt#1764]
Example code: https://godbolt.org/z/erD-_a

In v6 `fmt::format_to_n()` seems to work with `fmt::compile()`-d format string. 
In v7 though, just `fmt::format_to()`  is working with `FMT_COMPILE()`.

**Comment by bgemmill** [comment]
A workaround for now is to use fmt's truncating iterator with `fmt::format_to`:

```
  static const auto formatter = FMT_COMPILE("{}");
  std::array<char, 512> s2{0};

// These two lines should be equivalent
//  auto it = fmt::format_to_n(s0.data(), 512, formatter, 42).out; // fails at compile time
  auto it = fmt::format_to(fmt::detail::truncating_iterator<char*>(s2.data(), s2.size()), formatter, 42).base();

  std::cout << s2.data() << std::endl;
```

**Comment by vitaut** [comment]
A PR to add `format_to_n` overload that accepts `FMT_COMPILE` is welcome.

**Comment by vitaut** [comment]
Implemented in #1869.
